galaxyproject/galaxy
# [23.1] Workaround for XML nodes of job resource parameters losing their children

This is a workaround for, and thus closes #16727. 

This bug was spotted because the job resource parameters selector broke on usegalaxy.eu. For the tool keras_train_and_eval, the option "Use GPU resources" shows up, but no drop-down menu is available and no items can be selected.

The options are available as children of the XML nodes tagged with "param" within the scope of `galaxy.jobs.parse_resource_parameters`. They are then returned and later retrieved within the scope of `galaxy.util.parse_resource_parameters`, but by then, the children of the XML node (the options) are gone.

It seems that returning strings rather than lxml Element objects, and then converting the strings back to Element objects fixes the bug. I have no clue why. As said on #16727, this is likely not a Galaxy bug but rather something related either to lxml or whatever is doing the garbage collection. Nevertheless, this cannot be left broken.

## How to test the changes?
(Select all options that apply)
  1. Apply the changes of this PR.
  2. Visit the [keras_train_and_eval tool page](https://usegalaxy.eu/?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fbgruening%2Fkeras_train_and_eval%2Fkeras_train_and_eval%2F1.0.10.0&version=latest).
  3. Things work as expected again.

## License
